### PR TITLE
Changes to allow run with Jupyter Server 2

### PR DESCRIPTION
**Pull request by joaopamaral**
This PR adds some changes to allow run the `yarnspawner` with jupyter server version 2.

### CODE REVIEW

**Comment by Khrol** [comment]
@joaopamaral maybe merge everything (including my changes) into `master` branch? 

I'm not sure if @jcrist is going to support the original project. It's quite old already.

**Comment by joaopamaral** [comment]
> @joaopamaral maybe merge everything (including my changes) into `master` branch?
> 
> I'm not sure if @jcrist is going to support the original project. It's quite old already.

Yes, makes sense. I've changed the base branch of this PR to master.

File: yarnspawner/singleuser.py [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
 from traitlets import default
 
+if not os.environ.get("JUPYTERHUB_SINGLEUSER_APP"):
+    # setting this env prior to import of jupyterhub.singleuser avoids unnecessary import of notebook
+    os.environ["JUPYTERHUB_SINGLEUSER_APP"] = "jupyter_server.serverapp.ServerApp"
+
+try:
+    from jupyterhub.singleuser.mixins import make_singleuser_app
+except ImportError:
+    # backward-compat with jupyterhub < 1.3
```
Comment (Khrol) [inline yarnspawner/singleuser.py]: What the point if you've added `'jupyterhub>=4.0.0'`?

**Review by Khrol — COMMENTED** [review]
Small update about compatibility and LGTM

File: yarnspawner/singleuser.py (hunk 1 above)
Comment (joaopamaral) [reply yarnspawner/singleuser.py]: Good point, let me remove this part.

**Pull request merged**